vllm/model_executor/models/llava_onevision.py

```diff
@@ -793,11 +793,15 @@ def _process_video_pixels(self, inputs: LlavaOnevisionVideoPixelInputs):
 
         video_pixels = inputs["data"]
 
-        # TODO: support multiple videos per input
         if isinstance(video_pixels, torch.Tensor):
             stacked_embeddings = self._video_pixels_to_features(
                 self.vision_tower, video_pixels)
             return stacked_embeddings
+        elif is_list_of(video_pixels, torch.Tensor):
+            stacked_pixels = torch.cat(video_pixels, dim=0).unsqueeze(0)
+            stacked_embeddings = self._video_pixels_to_features(
+                self.vision_tower, stacked_pixels)
```

Comment:
I think we should convert them into embeddings first before combining them, in case the videos have different aspect ratios in the same batch.

Reply:
I think you are right.  But in LLaVA series models, the input videos or images have been pre-processed  same resolution or size. Converting into embedding first may result in a degradation of inference performance.

Reply:
In LLaVA-NeXT, the number of patches for each image can differ, resulting in different input tensor shapes per example in the batch. Is this not also the case for LLaVA-OneVision?